pkg/controller/provider/container/vsphere/watch.go

```diff
@@ -470,16 +470,19 @@ func (r *HostEventHandler) validate(host *model.Host) {
 	defer func() {
 		_ = tx.End()
 	}()
-	for _, ref := range host.Vms {
-		vm := &model.VM{}
-		vm.WithRef(ref)
-		err = tx.Get(vm)
-		if err != nil {
-			r.log.Error(err, "VM (get) failed.")
-			return
-		}
+	list := []model.VM{}
+	err = tx.List(
```

Comment:
Is this list restricted to just the VMs with references back to the host? It looks like it's querying all of them.

Reply:
Fixed.